lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -559,18 +468,19 @@ bool DiagnosticVerifier::verifyFile(unsigned BufferID,
 
     if (I == CapturedDiagnostics.end()) continue;
 
-    if (I->getMessage().find(expected.MessageStr) == StringRef::npos) {
+    auto columnNumber = SM.getLineAndColumn(I->Loc).second;
```

Comment:
We definitely want to use the column number from the diagnostic instance.  `llvm::SourceManager::getLineAndColumn` is going to go scanning buffers again.

Reply:
Good catch! We still need to call it once in `handleDiagnostic`, but that's equivalent to the scan `llvm::SMDiagnostic` would have done. I ran the test suite with both master and this branch and didn't notice any performance regressions.